From 7c386cbe38fc42c036bd14b9048b13f1a1d45877 Mon Sep 17 00:00:00 2001
From: Leah Rowe <info@minifree.org>
Date: Sun, 1 Dec 2024 23:40:08 +0000
Subject: [PATCH 1/1] t480 hack: turn off the dgpu

too buggy, in mkukri's testing. just turn it off.

thanks go to mkukri for showing me how to turn this off.
as nicholas explained to me, if there's no nvram on this board,
coreboot can't use a static option table, because that works by
copying those settings to the nvram.

so we just have to hardcode for the time being

Signed-off-by: Leah Rowe <info@minifree.org>
---
 3rdparty/vboot                                  | 2 +-
 src/mainboard/lenovo/sklkbl_thinkpad/ramstage.c | 4 +++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/3rdparty/vboot b/3rdparty/vboot
index f1f70f46dc..1f53ea9b7f 160000
--- a/3rdparty/vboot
+++ b/3rdparty/vboot
@@ -1 +1 @@
-Subproject commit f1f70f46dc5482bb7c654e53ed58d4001e386df2
+Subproject commit 1f53ea9b7f398884f722fca046129eae5ea6a71c
diff --git a/src/mainboard/lenovo/sklkbl_thinkpad/ramstage.c b/src/mainboard/lenovo/sklkbl_thinkpad/ramstage.c
index 9526642c57..bf5da12689 100644
--- a/src/mainboard/lenovo/sklkbl_thinkpad/ramstage.c
+++ b/src/mainboard/lenovo/sklkbl_thinkpad/ramstage.c
@@ -31,7 +31,9 @@ void mainboard_silicon_init_params(FSP_SIL_UPD *params)
 		dgfx_vram_id = gpio_get(GPIO_DGFX_VRAM_ID0) | gpio_get(GPIO_DGFX_VRAM_ID1) << 1;
 		printk(BIOS_DEBUG, "Discrete GPU present with %s VRAM\n", dgfx_vram_id_str[dgfx_vram_id]);
 
-		if (get_uint_option("dgpu_enable", 1)) {
+//		if (get_uint_option("dgpu_enable", 1)) {
+		if (get_uint_option("dgpu_enable", 0)) { // HACK: no option table due to no nvram.
+						// (turn off the dgpu by default due to bugs)
 			printk(BIOS_DEBUG, "Enabling discrete GPU\n");
 			// NOTE: i pulled this GPU enable sequence from thin air but it seems to work
 			gpio_set(GPIO_1R8VIDEO_AON_ON, 1);	// Enable GPU power rail
-- 
2.39.5

